File: list-models.js

require('dotenv').config({ path: '.env.local' });
const { GoogleGenerativeAI } = require('@google/generative-ai');

async function listModels() {
    const genAI = new GoogleGenerativeAI(process.env.GOOGLE_API_KEY);
    try {
        console.log("Listing models...");
        const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });
        const result = await model.generateContent("Hello");
        console.log("Flash response:", result.response.text());
        console.log("gemini-1.5-flash is AVAILABLE");
    } catch (e) {
        console.error("Flash failed:", e.message);
    }

    try {
        const model = genAI.getGenerativeModel({ model: "gemini-1.5-pro" });
        const result = await model.generateContent("Hello");
        console.log("Pro response:", result.response.text());
        console.log("gemini-1.5-pro is AVAILABLE");
    } catch (e) {
        console.error("Pro failed:", e.message);
    }
}

listModels();

File: list_models.py
import google.generativeai as genai
import os
from dotenv import load_dotenv

load_dotenv()
api_key = os.environ.get("GEMINI_API_KEY")
genai.configure(api_key=api_key)

try:
    with open("models.log", "w") as f:
        for m in genai.list_models():
            if 'generateContent' in m.supported_generation_methods:
                f.write(f"{m.name}\n")
    print("Models listed in models.log")
except Exception as e:
    with open("models.log", "w") as f:
        f.write(str(e))
